plugins/package-managers/node/src/main/kotlin/utils/NpmSupport.kt

```diff
@@ -153,16 +153,25 @@ internal fun parseNpmVcsInfo(node: JsonNode): VcsInfo {
         revision = head
     )
 
-    val type = repository["type"].textValueOrEmpty()
-    val url = repository.textValue() ?: repository["url"].textValueOrEmpty()
-    val path = repository["directory"].textValueOrEmpty()
-
-    return VcsInfo(
-        type = VcsType.forName(type),
-        url = expandNpmShortcutUrl(url),
-        revision = head,
-        path = path
-    )
+    return if (repository.isTextual) {
+        VcsInfo(
+            type = VcsType.UNKNOWN,
+            url = expandNpmShortcutUrl(repository.textValue()),
+            path = "",
+            revision = head
+        )
+    } else {
+        val type = repository["type"].textValueOrEmpty()
+        val url = repository.textValue() ?: repository["url"].textValueOrEmpty()
```

Comment:
I believe the original `repository.textValue()` here already handles the `repository.isTextual` case, or?

At least I'm pretty sure that original code already intended to fully handle that case:
- `type` would be empty, so `VcsType.forName(type)` returns `VcsType.UNKNOWN`
- `path` would be empty as well

So what additional case does your new code cover?

Reply:
HM, are you sure that the previous code already worked in this case?

When I wrote this code (in my other PR) I do re-call that the analyzer previously did not detect VCS info but afterwards did. So, thought it was needed but probably I made a mistake during testing and had made further changes which made it work.

Reply:
Test proofs that you were right.